packages/fxa-settings/src/pages/Index/container.tsx

```diff
@@ -49,25 +47,76 @@ export const IndexContainer = ({
   const isWebChannelIntegration =
     integration.isSync() || integration.isDesktopRelay();
 
-  // 'email' query param followed by 'login_hint' should take precedence
-  const email =
+  // 'email' query param should take precedence, followed by 'login_hint'
+  const suggestedEmail =
     queryParamModel.email ||
     integration.data.loginHint ||
     currentAccount()?.email;
-  const shouldRedirectToSignin = email && !prefillEmail;
+
+  const hasEmailSuggestion = suggestedEmail && !prefillEmail;
+
+  const handleNavigation = useCallback(
+    (
+      exists: boolean,
+      hasLinkedAccount: boolean,
+      hasPassword: boolean,
+      email: string
+    ) => {
+      const params = new URLSearchParams(location.search);
+      // We delete 'email' if it is present because otherwise, that param will take
+      // precedence and the user will be unable to create an account with a different
+      // email. Remove for signin as well as it is unnecessary. This is a byproduct
+      // of backwards compatibility between Backbone and React since we pass this
+      // param from content-server, TODO: FXA-10567
+      // We can also use useNavigateWithQuery after addressing the above.
+      params.delete('email');
+      const hasParams = params.size > 0;
+      const isOAuth = location.pathname.startsWith('/oauth');
+      if (!exists) {
+        const signupRoute = isOAuth ? '/oauth/signup' : '/signup';
+        console.log(
+          `${signupRoute}${hasParams ? `?${params.toString()}` : ''}`
+        );
+        navigate(`${signupRoute}${hasParams ? `?${params.toString()}` : ''}`, {
```

Comment:
If params is empty, toString() will result in '' anyway, so I don't see the need for the ternary operator.

Reply:
Updated to use the URL API instead of this ternary.